superset/views/base.py

```diff
@@ -170,16 +170,61 @@ def json_response(self, obj, status=200):
             mimetype="application/json",
         )
 
+    def menu_data(self):
+        menu = appbuilder.menu.get_data()
+        root_path = "#"
+        if not g.user.is_anonymous:
+            try:
+                logo_target_path = appbuilder.app.config.get(
+                    "LOGO_TARGET_PATH"
+                ) or "/profile/{}/".format(g.user.username)
+                if logo_target_path.startswith("/"):
+                    root_path = "/superset{}".format(logo_target_path)
+                else:
+                    root_path = logo_target_path
+            except NameError as e:
```

Comment:
Which line is likely to raise a `NameError`? Can we target it?

Reply:
it's in the case `user.username` is not set, left a comment.